pure-selenium/src/main/java/org/example/pages/LoginPage.java

```diff
@@ -18,15 +15,33 @@ public class LoginPage {
     @FindBy(id = "login-button")
     private WebElement loginButton;
 
+    @FindBy(css = "h3[data-test='error']")
+    private WebElement errorMessage;
+
+    public boolean isErrorMessageDisplayed() {
+        return errorMessage.isDisplayed();
+    }
+
+    public String getErrorMessageText() {
+        return errorMessage.getText();
+    }
 
     public LoginPage(WebDriver driver) {
-        this.driver = driver;
-        PageFactory.initElements(driver, this);
+        super(driver);
     }
 
-    public void login(String username, String password) {
+    public void open() {
+        driver.get("https://www.saucedemo.com/");
```

Comment:
Put the URL in the configuration file, which significantly improves the flexibility and maintainability of the project